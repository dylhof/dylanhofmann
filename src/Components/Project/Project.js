import React from 'react';

export const Project = ({ title, git, url, description, thumbnail}) => {
  return(
    <div className='project-div'>
      <h2>{title}</h2>
      <img src={thumbnail} alt={title}/>
      <p className='project-description-p'>{description}</p>
      <p>gitHub link: <a href={git} target='blank'>{git}</a></p>
      <p>Live link: <a href={url} target='blank'>{url}</a></p>
    </div>
  )
};
