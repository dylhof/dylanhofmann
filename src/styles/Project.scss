.projectArea-div{
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(380px, 1fr));
}

.project-div {
  width: 370px;
  height: 555px;
  border: 1px black solid;
  margin: 50px;

  h2 {
    @include text(20px, 700);
    margin: 10px;
  }

  img {
    max-height: 165px;
    max-width: 98%;
    border: 1px black solid;
    overflow: scroll;
  }

  .project-description-p {
    height: 200px;
    margin: 10px;
  }
}