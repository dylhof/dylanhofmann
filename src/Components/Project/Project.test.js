import React from 'react';
import { Project } from './Project';
import { shallow } from 'enzyme';

describe('Project component', () => {
  let wrapper;
  const mockproject = {
    title: 'mockproject',
    git: 'www.mockgit.com',
    url: 'www.mockproject.com',
    description: 'this is a mock project',
    thumbnail: 'www.thumbnail.com'
  }
  beforeEach(() => {
    wrapper = shallow(<Project {...mockproject}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});