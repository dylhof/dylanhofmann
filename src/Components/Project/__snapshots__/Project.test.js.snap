// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component should match the snapshot 1`] = `
<div
  className="project-div"
>
  <h2>
    mockproject
  </h2>
  <img
    alt="mockproject"
    src="www.thumbnail.com"
  />
  <p
    className="project-description-p"
  >
    this is a mock project
  </p>
  <p>
    gitHub link: 
  </p>
  <a
    href="www.mockgit.com"
    target="blank"
  >
    www.mockgit.com
  </a>
  <p>
    Live link: 
  </p>
  <a
    href="www.mockproject.com"
    target="blank"
  >
    www.mockproject.com
  </a>
</div>
`;
